ui_components/src/village/VillageUpgrades.js

```diff
@@ -0,0 +1,754 @@
+                                                    header: 'Confirmation',
+                                                    text: "Are you sure you want to cancel construction of " + selectedBuilding.name + "?\nExisting progress toward construction will be saved.",
+                                                    ContentComponent: null,
+                                                    onConfirm: () => CancelConstruction(),
+                                                })}>cancel</div>
+                                                {selectedBuilding.construction_boosted !== "true"
+                                                    ?
+                                                    <div className="construction_boost_button upgrades_control_button" onClick={() => CheckBoostConstruction()}>boost</div>
+                                                    :
+                                                    <div className="construction_boost_button upgrades_control_button disabled">boost</div>
+                                                }
+
+                                            </>
+                                        }
+                                        {selectedBuilding.status !== "upgrading" &&
+                                            <>
+                                                {(selectedBuilding.health < selectedBuilding.max_health)
+                                                    ?
+                                                    <>
+                                                        <div className="repair_begin_button upgrades_control_button" onClick={() => openModal({
+                                                            header: 'Confirmation',
+                                                            text: "sometext?",
+                                                            ContentComponent: null,
+                                                            onConfirm: () => BeginRepairs(),
+                                                        })}>repair</div>
+                                                    </>
+                                                    :
+                                                    <>
+                                                        <div className="repair_begin_button upgrades_control_button disabled">repair</div>
+                                                    </>
+                                                }
+                                            </>
+                                        }
+                                    </div>
+                                }
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div> 
+        );
+    }
+    const renderUpgradesContainer = () => {
```

Comment:
Same comment here, this probably should be pulled out into its own component? And potentially some of the sub `render` functions as well